### ISSUE TITLE
[$175] Additions to `Send to Topcoder` menu

### ISSUE THREAD

**Issue by jmgasper** [topcoder-platform/tcx-azure-devops-ext#26]
@afrisalyp - New details from Topcoder:

so I think, for the initial phase, we're going to put it into a draft challenge and let the CP or CA then review, refine, and set prizes. We can think how to get more TCX-like automatic as we go. 

### New feature:

Would it be possible to build a screen before sending the data to the API before that that auto fills w/ the task details to be sent over first?

maybe we can add prizes on that screen as an option

**Comment by jmgasper** [comment]
Contest https://www.topcoder.com/challenges/30143671 has been created for this ticket.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by jmgasper** [comment]
Contest https://www.topcoder.com/challenges/30143671 has been updated - it has been assigned to afrisalyp.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by afrisalyp** [comment]
@jmgasper 

It seems not possible to show a view from the menu action. The views need to be inside our extension iframe (e.g. Widget, hub, or work item page).

Alternatively, we can consider to put the feature inside a work item page. Please refer to this link

https://docs.microsoft.com/en-us/azure/devops/extend/develop/add-workitem-extension?view=azure-devops#add-a-page

**Comment by jmgasper** [comment]
@afrisalyp - Thanks.  I'll ask Topcoder what their preference is.

**Comment by jmgasper** [comment]
@afrisalyp - Feedback from Topcoder:

That seems like a good option!
Let’s try that and then do a drop for the client and we can get some feedback

If I need to adjust the payment higher, let me know, thanks.

**Comment by jmgasper** [comment]
This ticket was not processed for payment. If you would like to process it for payment, please reopen it, add the ```tcx_FixAccepted``` label, and then close it again<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by afrisalyp** [comment]
@jmgasper 

A Pull request sent for this new feature.
It will look like:

![Screenshot from 2020-09-28 12-03-52](https://user-images.githubusercontent.com/8887344/94392591-e7b97780-0182-11eb-9f99-5b4134183006.png)

**Comment by jmgasper** [comment]
@afrisalyp - Thanks, this looks good.  A couple things here:

1.  `Topcoder Challenge Id` will be set by the system, let's not have that as editable
2.  If an error occurs, can we show a popup or something to alert the user?  I got a 403 from the V5 API on my first test, but I got no indication in the UI
3.  We don't have to open up the challenge on the Topcoder site
4.  The challenge ID we receive should be used to populate the field on the work item:

![Screen_Shot_2020-09-28_at_15_46_31](https://user-images.githubusercontent.com/21790/94395048-18f57000-01a2-11eb-88f8-4618e1202c4a.png)
